pyccl/pk2d.py

```diff
@@ -111,18 +125,26 @@ def __init__(self, pkfunc=None, a_arr=None, lk_arr=None, pk_arr=None,
                 pkflat[ia, :] = pkfunc(k=np.exp(lk_arr), a=a)
             pkflat = pkflat.flatten()
 
-        self.extrap_order_lok = extrap_order_lok
-        self.extrap_order_hik = extrap_order_hik
-
         self.psp, status = lib.set_pk2d_new_from_arrays(lk_arr, a_arr, pkflat,
                                                         int(extrap_order_lok),
                                                         int(extrap_order_hik),
                                                         int(is_logp), status)
         check(status, cosmo=cosmo)
-        self.has_psp = True
+
+    @property
+    def has_psp(self):
+        return 'psp' in vars(self)
+
+    @property
+    def extrap_order_lok(self):
+        return self.psp.extrap_order_lok
```

Comment:
It doesn't make sense to create an empty `Pk2D` to fill it out later (on the user side). The keyword `empty` only exists because we create empty `Pk2D` objects internally whenever we call a linear or non-linear model. So, it is guaranteed it will always contain a `psp`.

You can create empty (empty object dict) objects in Pyhton using `ClassName.__new__(ClassName)`. This should have been the case all along, I don't know why the `empty` keyword even exists. Shall I just remove it?

Reply:
Ok, I see your point, but removing `empty` will break the API. Do you want to leave that for another PR?

Reply:
(either way, I think it's ok to leave the line of code that this thread was about, unchanged)

Reply:
Okay, let's leave it for another PR.